### PR TITLE
Remove API codepaths < 1.12

### PR DESCRIPTION
**Pull request by runcom**
Quite sure everything is ok but I think I could still miss something (for instance, removing `Json` field from `Image(.*)Config`)

/cc @tiborvass @LK4D4  

Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -752,7 +723,7 @@ func (s *Server) postImagesCreate(version version.Version, w http.ResponseWriter
 		if !output.Flushed() {
 			return err
 		}
-		sf := streamformatter.NewStreamFormatter(useJSON)
+		sf := streamformatter.NewStreamFormatter(true)
```
Comment (LK4D4) [inline api/server/server.go]: Is somewhere this used without `json`?

File: api/server/server.go [hunk 2]
```diff
@@ -717,15 +689,14 @@ func (s *Server) postImagesCreate(version version.Version, w http.ResponseWriter
 		}
 
 		imagePullConfig := &graph.ImagePullConfig{
-			Parallel:    version.GreaterThan("1.3"),
+			Parallel:    true,
```
Comment (tiborvass) [inline api/server/server.go]: in fact you can remove Parallel altogether, it's equivalent to true wherever it's referenced.

File: api/server/server.go [hunk 3]
```diff
@@ -717,15 +689,14 @@ func (s *Server) postImagesCreate(version version.Version, w http.ResponseWriter
...
 			MetaHeaders: metaHeaders,
 			AuthConfig:  authConfig,
 			OutStream:   output,
-			Json:        useJSON,
+			Json:        true,
```
Comment (tiborvass) [inline api/server/server.go]: same for this

File: api/server/server.go (hunk 1 above)
Comment (tiborvass) [inline api/server/server.go]: also if nobody uses NewStreamFormatter with `false`, you can remove the param to the function and refactor that package

**Comment by LK4D4** [comment]
Probably we can make StreamFormatter always json.

**Comment by runcom** [comment]
@LK4D4 @tiborvass alright cool I already did what you both said in another branch but wasn't sure if this pr should address what you said

**Comment by LK4D4** [comment]
@runcom Yeah, it should :) Also you already need rebase.

**Comment by runcom** [comment]
@LK4D4 @tiborvass updated

**Comment by jfrazelle** [comment]
eeee <3

File: api/server/server.go (hunk 1 above)
Comment (runcom) [inline api/server/server.go]: I'll address this in another pr

File: api/server/server.go [hunk 4]
```diff
@@ -1255,10 +1180,8 @@ func (s *Server) postBuild(version version.Version, w http.ResponseWriter, r *ht
 		}
 	}
 
-	if version.GreaterThanOrEqualTo("1.8") {
-		w.Header().Set("Content-Type", "application/json")
-		buildConfig.JSONFormat = true
-	}
+	w.Header().Set("Content-Type", "application/json")
+	buildConfig.JSONFormat = true
```
Comment (LK4D4) [inline api/server/server.go]: This field is needed for StreamFormatter too.

Comment (runcom) [inline api/server/server.go]: yup I'm updating the builder just now (didn't realize)

**Comment by runcom** [comment]
@jfrazelle :metal: 
Btw I'm pretty sure I removed everything, I'll make another pr addressing hardcoded bool in `NewStreamFormatter`, do you guys think it still misses something or I forgot some path?

**Comment by jfrazelle** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Comment by HuKeping** [comment]
I'm pretty mush like this PR, finally we get ride of those version checks thanks!

**Pull request merged**